examples/plot_events.py

```diff
@@ -1,75 +1,79 @@
 import matplotlib.pyplot as plt
 import numpy as np
-from pkg_resources import resource_filename
+from argparse import ArgumentParser
+from functools import lru_cache
 
 import astropy.units as u
 
 from ctapipe.instrument import CameraGeometry
 from ctapipe.visualization import CameraDisplay
 
-from eventio import EventIOFile
-from eventio.simtel import (
-    CameraSettings,
-    Event,
-    TelescopeEvent,
-    ADCSamples
-)
-
-input_file = resource_filename(
-    'eventio',
-    'gamma_test.simtel.gz',
-)
-input_file = '/home/maxnoe/Downloads/gamma_20deg_180deg_run7360___cta-prod3-merged_desert-2150m-Paranal-3HB89-NGFD.simtel.gz'
-
-
-with EventIOFile(input_file) as f:
-    cameras = {}
-    for o in f:
-        if isinstance(o, CameraSettings):
-            cam_data = o.parse()
-
-            if cam_data['pixel_shape'][0] == 2:
-                pix_type = 'square'
+from eventio.simtel import SimTelFile
+
+
+parser = ArgumentParser()
+parser.add_argument('inputfile')
+args = parser.parse_args()
+
+
+@lru_cache()
+def build_cam_geom(simtel_file, telescope_id):
+    cam_data = simtel_file.telescope_descriptions[telescope_id]['camera_settings']
+
+    if cam_data['pixel_shape'][0] == 2:
+        pix_type = 'square'
+        pix_rotation = 0 * u.deg
+
+    elif cam_data['pixel_shape'][0] == 1:
+        pix_type = 'hexagonal'
+
+        # LST has 0 deg rotation, MST 30 (flat top vs. pointy top hexagons)
+        if cam_data['n_pixels'] == 1855:
+            pix_rotation = 0 * u.deg
+        else:
+            pix_rotation = 30 * u.deg
+
+    # if pix_type == -1, we have to guess
+    elif cam_data['pixel_shape'][0] == -1:
+        if cam_data['n_pixels'] > 2000:
+            pix_type = 'square'
+            pix_rotation = 0 * u.deg
+        else:
+            pix_type = 'hexagonal'
+
+            # LST has 0 deg rotation, MST 30 (flat top vs. pointy top hexagons)
+            if cam_data['n_pixels'] == 1855:
                 pix_rotation = 0 * u.deg
+            else:
+                pix_rotation = 30 * u.deg
+
+    return CameraGeometry(
+        cam_id='CAM-{}'.format(telescope_id),
+        pix_id=np.arange(cam_data['n_pixels']),
+        pix_x=cam_data['pixel_x'] * u.m,
+        pix_y=cam_data['pixel_y'] * u.m,
+        pix_area=cam_data['pixel_area'] * u.m**2,
+        pix_type=pix_type,
+        cam_rotation=cam_data['cam_rot'] * u.rad,
+        pix_rotation=pix_rotation,
+    )
+
```

Comment:
I think it is great, that `build_cam_geom` was broken out of the previously very long script. This gives us a better possibility to test this function.